test/unit_test/workflows/mock_data/text_generation_dataset_random.json

```diff
@@ -0,0 +1,62 @@
+{
+    "input_model":{
+        "type": "PyTorchModel",
+        "config": {
+            "hf_config": {
+                "model_name": "gpt2",
+                "task": "text-generation",
+                "dataset": {
+                    "data_name": "ptb_text_only",
+                    "subset": "penn_treebank",
+                    "split": "train",
+                    "input_cols": ["sentence"],
+                    "component_kwargs": {
```

Comment:
in this example, I don't see the params_config is demonstrated in the PR description. How does it look like if both params_config and component_kwargs exist at the same time?

Reply:
This is equivalent to
```
{
    "name": "ptb",
    "type": "HuggingfaceContainer",
    "params_config": {
        "model_name": "gpt2",
        "task": "text-generation",
        "data_name": "ptb_text_only",
        "subset": "penn_treebank",
        "split": "train"
        "input_cols": ["sentence"],
        "component_kwargs": {
            "pre_process_data": {
                "seqlen": 1024,
                 "max_samples": 5,
                 random_seed": 42
            }
        }
    }
  }
```
Similar to the one shown in the PR description. olive.workflows.run automatically converts hf_config.dataset to this format.

Reply:
It would be something like 
```
"raw_data": {
            "name": "raw_data",
            "type": "RawDataContainer",
            "user_script": "user_script.py",
            "components": {
                "post_process_data": {
                    "type": "inception_post_process"
                }
            },
            "params_config": {
                "data_dir": "data",
                "component_kwargs: {
                     "load_data": {
                        "input_names": ["input"],
                         "input_shapes": [[1, 299, 299, 3]],
                         "input_order_file": "input_order.txt",
                         "annotations_file": "labels.npy",
                      }
                 },
               "batch_size": 7
        }
```
But this is not necessary since all of the `load_data` args are name arguments in https://github.com/microsoft/Olive/blob/main/olive/data/component/load_dataset.py#L46. So they can be added directly to `params_config`. 
`component_args` is most useful when we want to provide values to `kwargs` in a function like https://github.com/microsoft/Olive/blob/main/olive/data/component/load_dataset.py#L26C101-L26C109.

Reply:
If we'd like to pass parameters to components.post_process_data in above example, need we define two sections? one for components and another for component_kwargs? If yes, it would be better if we could put all the definitions including  post_process_data and the params in consolidated place instead of spreading them.

Reply:
We can pass the parameters directly to the components.post_process_data.params. It is equivalent to passing it to component_kwargs. So two sections are not needed for the same component `post_process`. 
`component_kwargs` is just an additional way of providing kwargs without creating a `components` dict which is not possible for pytorch_model.hf_config.

Reply:
Resolved offline.